examples/plugin/wasmedge-zlib/main.cpp

```diff
@@ -0,0 +1,238 @@
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <iterator>
+#include <random>
+#include <type_traits>
+#include <vector>
+
+// ----------------------------------------------------------
+// TODO: LOOK FOR A .WIT REPLACEMENT
+
+#ifdef __EMSCRIPTEN__
+#include <emscripten.h>
+#define PRESERVE EMSCRIPTEN_KEEPALIVE
+#define IMPORT __attribute__((import_module("wasmedge_zlib"))) extern "C"
+#else
+#include <zlib.h>
+#define PRESERVE
+#define IMPORT extern "C"
+#endif
+
+#ifdef __EMSCRIPTEN__
+
+#define ZLIB_VERSION "1.2.11"
+#define Z_OK 0
+#define Z_STREAM_END 1
+#define Z_STREAM_ERROR (-2)
+#define Z_FINISH 4
+#define Z_NULL NULL
+
+typedef void *(*alloc_func)(void *opaque, unsigned int items,
+                            unsigned int size);
+typedef void (*free_func)(void *opaque, void *address);
+
+struct internal_state;
+
+typedef struct z_stream_s {
```

Comment:
Is there any reason not to apply the coding style to this structure?

Reply:
@hydai 
Ideally the user should not be writing these zlib internal structures.

I think the ideal way here will be to provide a `.wit` file to describe all the zlib structures & all.
Then use [wit-bindgen](https://github.com/bytecodealliance/wit-bindgen) to create these structures on the fly.

I created this PR, to provide an example to the `wasmedge-zlib` for the time being.


So the usage of `WASM Components` should be the ideal